src/swish/app.ms

```diff
@@ -199,6 +199,72 @@
     (printf "~{~a\n~}" stderr)
     (raise 'matched-wrong-case)]))
 
+(isolate-mat default-signal-handlers ()
+  (define-values (to-stdin from-stdout from-stderr os-pid)
+    (match (osi_get_argv)
+      [#()
+       (parameterize ([cd "src"])
+         (spawn-os-process scheme-exe
+           '("--eedisable" "--program" "repl.ss" "-q" "--")
+           self))]
+      [,_
+       (spawn-os-process (osi_get_executable_path) '("-q" "--") self)]))
+  (let ([to-stdin (binary->utf8 to-stdin)]
+        [from-stdout (binary->utf8 from-stdout)]
+        [from-stderr (binary->utf8 from-stderr)]
+        [me self])
+    (define (dump p)
+      (let ([line (get-line p)])
+        (unless (eof-object? line)
+          (send me (cons p line))
+          (dump p))))
+    (spawn&link (lambda () (dump from-stderr)))
+    (spawn&link (lambda () (dump from-stdout)))
+    (pretty-print
+     ;; single expression to avoid gatekeeper interaction after app:start
+     `(begin
+        (app:start)
+        (printf "started\n")
+        (flush-output-port)
+        (receive (after 1000) (exit 3)))
+     to-stdin)
+    (flush-output-port to-stdin)
+    (match-let*
+     (["started"
+       (receive (after 1500 'bad-start)
+         [(,@from-stdout . ,msg) msg])]
+      [,!
+       (meta-cond
+        [(memq (machine-type) '(i3nt ti3nt a6nt ta6nt))
+         ;; osi_kill won't work for this on Windows, but I haven't
+         ;; gotten GenerateConsoleCtrlEvent to work because I gave
+         ;; it a dwProcessGroupId that tore down too much.
+         (raise 'need-help-from-Bob)]
```

Comment:
@Bob, is there some way to automate this test or will we have to test this manually?

Reply:
I don't know how to automate this test on Windows, so I'd leave it as a manual test.